runner.go

```diff
@@ -0,0 +1,104 @@
+package async
+
+import (
+	"context"
+	"sync"
+)
+
+type Runner struct {
+	sync.Mutex
+	tasks      []Task
+	errs       []error
+	limit      int
+	waitErrors bool
+}
+
+// NewRunner creates a new task manager to control async functions.
+func NewRunner(tasks ...Task) *Runner {
+	return &Runner{
+		tasks: tasks,
+		limit: len(tasks),
+	}
+}
+
+// WaitErrors tells the runner to wait for the response from all functions instead of cancelling them all when the first error occurs.
+func (r *Runner) WaitErrors() *Runner {
+	r.waitErrors = true
+	return r
+}
+
+// WithLimit defines a limit for concurrent tasks execution
+func (r *Runner) WithLimit(limit int) *Runner {
+	r.limit = limit
+	return r
+}
+
+// AllErrors returns all errors reported by functions
+func (r *Runner) AllErrors() []error {
```

Comment:
Retornar todos os erros no mesmo obj que responda a interface de erro, e depois tratar usando `error.As` não seria mais idiomático? 
```go
err := async.NewRunner(func(context.Context) error {
    return errors.New("1")
}, 
func(context.Context) error {
    return errors.New("2")
}).Run(ctx)
var aerr *async.MultiErr
if errors.As(err, &aerr) {
    for _, err := range aerr { // ... }
}

```

Reply:
O comportamento atual retorna o primeiro erro que ocorrer sem retornar ele. Nesse formato o `err == ErrExample` não funcionaria mais. O usuário é obrigado a usar o `errors.Is` ou `errrors.As`. Vai quebrar muito código nosso, o que acha?

Reply:
esse `AllErrors` em tese seria apenas mais uma alternativa caso ele chamar o Runner com a opção `WaitErrors`